commit f190f6f06deb455ce73166b7a8810dc56bba1b88
Author: Keerthana K <keerthanak@vmware.com>
Date:   Fri Jun 19 12:55:56 2020 +0530

    custom script: Add DEFAULT-RUN-POST-CUST-SCRIPT

    DEFAULT-RUN-POST-CUST-SCRIPT is added to cust.cfg. Custom Scripts are
    run based on the following conditions:
       1. If “enable-custom-script” is absent / not set, then
          1.1 If DEFAULT-RUN-POST-CUST-SCRIPT is yes, then run custom script.
          1.2 If DEFAULT-RUN-POST-CUST-SCRIPT is no or not set, then fail / abort Guest Customization.
       2. If “enable-custom-script” is explicitly set, then
          2.1 If “enable-custom-script” is set to true, then run custom script.
          2.2 If “enable-custom-script” is set to false, then fail / abort Guest Customization.

diff --git gosc-scripts/imc-shell/imgcust-scripts/ConfigFile.sh gosc-scripts-modify/imc-shell/imgcust-scripts/ConfigFile.sh
index 1d6e064..f773d2f 100755
--- gosc-scripts/imc-shell/imgcust-scripts/ConfigFile.sh
+++ gosc-scripts-modify/imc-shell/imgcust-scripts/ConfigFile.sh
@@ -1400,3 +1400,19 @@ ConfigFile_GetPrimaryNic()
   echo "$primary"
 }
 
+# Retrieves whether to force run post custom script.
+#
+# Args:
+#   None
+# Results:
+#   boolean: 1 for yes/YES, 0 for no/NO, an empty string if it wasn't specified
+# Throws:
+#   Dies in case setting is present, but is not yes/YES or no/NO.
+ConfigFile_GetForceRunPostCustScript()
+{
+  local val='' # has to be declared before assigned
+
+  val=$(ConfigFile_GetOptionalBoolean 'MISC|DEFAULT-RUN-POST-CUST-SCRIPT') || exit 1
+
+  echo "$val"
+}
diff --git gosc-scripts/imc-shell/imgcust-scripts/CustomizationUtils.sh gosc-scripts-modify/imc-shell/imgcust-scripts/CustomizationUtils.sh
index a851d88..0f3d052 100755
--- gosc-scripts/imc-shell/imgcust-scripts/CustomizationUtils.sh
+++ gosc-scripts-modify/imc-shell/imgcust-scripts/CustomizationUtils.sh
@@ -1305,8 +1305,10 @@ GetToolsConfig() {
       IFS=' ' read -r -a array <<< "$result"
       if echo "${array[3]}" | grep -iqwF "true"; then
          echo "${array[3]}"
+      elif echo "${array[3]}" | grep -iqwF "false"; then
+         echo "${array[3]}"
       else
-         echo "false"
+         echo "not set"
       fi
    else
       echo "false"
diff --git gosc-scripts/imc-shell/imgcust-scripts/PhotonCustomization.sh gosc-scripts-modify/imc-shell/imgcust-scripts/PhotonCustomization.sh
index 10d8795..822df82 100755
--- gosc-scripts/imc-shell/imgcust-scripts/PhotonCustomization.sh
+++ gosc-scripts-modify/imc-shell/imgcust-scripts/PhotonCustomization.sh
@@ -12,15 +12,28 @@ RunCloudConfig()
 
   ConfigFile_LoadConfigFile $configPath
 
-  #Check whether post custom script is present and the file is non empty then
-  #Read enable-custom-scripts value from /etc/vmware-tools/tools.conf through vmware-toolbox-cmd command
-  #If enable-custom-scripts value is false then cancel the guest customization and Report failure.
+  # Check whether post custom script is present and the file is non empty then
+  # Check "enable-custom-script" value from VM Tools configuration:
+       # If "enable-custom-script" is absent / not set, then
+           # If DEFAULT-RUN-POST-CUST-SCRIPT is yes, then run custom script.
+           # If DEFAULT-RUN-POST-CUST-SCRIPT is no or not set, then report error and abort the guest customization.
+       # If "enable-custom-script" is explicitly set, then enable/disable the post script per "enable-custom-scripts" value.
+           # If “enable-custom-script” is set to true, then run custom script.
+           # If “enable-custom-script” is set to false, then report error and abort the guest customization.
+
   local postScript=$(ConfigFile_GetCustomScriptName)
   if [[ -n "$postScript" ]]; then
      postScript=`${DIRNAME} $configFilePath`"/$postScript"
      if [[ -f "$postScript" && -s "$postScript" ]]; then
         local value=$(GetToolsConfig "deployPkg" "enable-custom-scripts")
-        if !(echo $value | grep -iqwF "true"); then
+        if (echo $value | grep -iqwF "not set"); then
+           local forceCustomScript=$(ConfigFile_GetForceRunPostCustScript)
+           Debug "DEFAULT-RUN-POST-CUST-SCRIPT: [$forceCustomScript]"
+           if [[ -z "$forceCustomScript" || $forceCustomScript == 0 ]]; then
+              PostDeployPkgStatus "enable-custom-scripts is not set and DEFAULT-RUN-POST-CUST-SCRIPT is no or not set"
+              Die "enable-custom-scripts is not set and DEFAULT-RUN-POST-CUST-SCRIPT is no or not set. Failing customization"
+           fi
+        elif (echo $value | grep -iqwF "false"); then
            PostDeployPkgStatus "enable-custom-scripts is set to false"
            Die "enable-custom-scripts is set to false. Failing customization"
         fi
